src_py/apiServer/decoderHttpMainServer.py

```diff
@@ -1,18 +1,62 @@
 
 from decoderHttpMainServerDefs import *
 
+# | seperates entities
+# & start entities stats
+# # seperates (stat_name-stat_value-type) -> triplets for entity without workers
+# : seperates stat_name , stat_value and type
+# ^ seperates workers belong to client 
+#  Only for workers:
+# @ seperates (stat_name, stat_value and type) -> worker_triplets for worker
+# $ seperates stat_name, stat_value and type
+
+# Todo: fix all examples in the function
 def decode_main_server_ets_str(string_to_convert: str):
-    result_list = []
+    result_dict = {} # {entity_name: ets_dict,...}
+
+    entity_with_stats_list = string_to_convert.split("|")[:-1] # Example: [c2&bytes_sent:0:int#messages_sent:8015:int#bad_messages:0:int#batches_sent:0:int#messages_received:8015:int#batches_received:0:int#bytes_received:40022:int#batches_dropped:0:int#messages_dropped:0:int#, w4^bytes_sent$0$int@empty_batches$0$int@bad_messages$0$int@batches_dropped_predict$0$int@batches_sent_predict$0$int@nan_loss_count$0$int@batches_received_predict$1001$int@batches_dropped_train$0$int@acc_time_training$0$int@bytes_received$0$int@average_time_prediction$0$int@batches_received_train$1000$int@average_time_training$0$int@acc_time_prediction$0$int@batches_sent_train$0$int@, s1&bytes_sent:0:int#messages_sent:4:int#bad_messages:0:int#batches_sent:8004:int#messages_received:7:int#batches_received:0:int#bytes_received:0:int#batches_dropped:0:int#messages_dropped:0:int#, c1&bytes_sent:54036:int#messages_sent:8016:int#bad_messages:0:int#batches_sent:0:int#messages_received:8016:int#batches_received:0:int#bytes_received:40022:int#batches_dropped:0:int#messages_dropped:0:int#, w2^bytes_sent$0$int@empty_batches$0$int@bad_messages$0$int@batches_dropped_predict$0$int@batches_sent_predict$0$int@nan_loss_count$0$int@batches_received_predict$1001$int@batches_dropped_train$0$int@acc_time_training$0$int@bytes_received$0$int@average_time_prediction$0$int@batches_received_train$1000$int@average_time_training$0$int@acc_time_prediction$0$int@batches_sent_train$0$int@, w1^bytes_sent$0$int@empty_batches$0$int@bad_messages$0$int@batches_dropped_predict$0$int@batches_sent_predict$0$int@nan_loss..]
+    for entity_with_stats in entity_with_stats_list:
+        # Example: entity_with_stats = c2&bytes_sent:0:int#messages_sent:8015:int#bad_messages:0:int#batches_sent:0:int...
+        if "&" in entity_with_stats:
+            # Example: entity_with_stats = c2&bytes_sent:0:int#messages_sent:8015:int#bad_messages:0:int#batches_sent:0:int...
+            entity_name = entity_with_stats.split('&')[0] # Example: c2
```

Comment:
@ohad123 @NoaShapira8 
Please use definitions for each separator (in the beginning of this file)
E.g.

SEP_ENTITY_AND_STATS='&'

Reply:
I added the definitions in decoderHttpMainServerDefs file